lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,253 @@
+ * limitations under the License.
+ */
+'use strict';
+
+require('./browser_api.es6');
+
+foam.CLASS({
+  name: 'WebAPIs',
+  package: 'org.chromium.apis.web',
+  documentation: `WebAPIs is the main model of this project. It has a
+    property called browserAPIs which is a FOAM DAO. It contains interface and
+    API information for major browsers and version for recent two years. It
+    has methods to transform the data model from Google Cloud Datastore
+    to CSV format String or JSON structure to be displayed in client side.
+    It also has methods to do data analysis.`,
+  requires: [
+    'org.chromium.apis.web.BrowserAPI',
+    'foam.dao.EasyDAO',
+    'foam.dao.ArrayDAO',
+    'foam.mlang.ExpressionsSingleton',
+    'foam.mlang.sink.GroupBy',
+    'foam.mlang.sink.Count',
+  ],
+  properties: [
+    {
+      name: 'browserAPIs',
+      documentation: `This should be a server GCD DAO that contains
+        interfaece, API information for major browsers and versions
+        for recent 2 years. It is implemented as a MDAO for now.`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'BrowserAPIDAO',
+          of: this.BrowserAPI,
+          daoType: 'MDAO',
+        }).addPropertyIndex(this.BrowserAPI.BROWSER_KEY)
+        .addPropertyIndex(this.BrowserAPI.INTERFACE_KEY);
+      },
+    },
+    {
+      name: 'mlang',
+      documentation: `FOAM's mlang expressiongs, it supports Math and
+        logic operations`,
+      factory: function() {
+        return this.ExpressionsSingleton.create();
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'importAPI',
+      documentation: `Anynchronous function to import interface/API for a
+        given version of browser. The function will do nothing if the given
+        browser already exists in browserAPIs`,
+      args: [
+        {
+          name: 'browserName',
+          documentation: `The name of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'browserVersion',
+          documentation: `The version of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'osName',
+          documentation: `The name of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'osVersion',
+          documentation: `The version of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'apiCatalog',
+          documentation: `The web catalog of this version of
+            browser, tested on the given operating system and version.
+            The json object should be the JSON object returned from
+            com.web.catalog.apiExtractor`,
+          typeName: 'JSON',
+        },
+      ],
+      code: function(browserName, browserVersion,
+        osName, osVersion, apiCatalog) {
+        let browserKey = `${browserName}_${browserVersion}_${osName}_${osVersion}`;
```

Comment:
There should be a shared implementation for constructing these keys. I think we should do one of two things here:

(1) Create a `foam.LIB` that implements `${browserName}_${browserVersion}_${osName}_${osVersion}`. Make the `browserKey` `expression` call out to the lib, and also call out to the lib here.
(2) At this point, construct a `BrowserAPI`, then ask for its `browserKey`.

Both solutions are fine by me. We may create more garbage with (2) though.